plugins/ui/src/js/src/widget/WidgetUtils.tsx

```diff
@@ -160,14 +161,20 @@ export function getComponentForElement(element: ElementNode): React.ReactNode {
     const Component = getComponentTypeForElement(newElement);
 
     if (Component != null) {
-      const props =
+      const { props } = newElement;
```

Comment:
I'd clone the props to avoid mutation, just in case.
```suggestion
      const props = { ...newElement.props };
```